### ISSUE TITLE
[Feature Request] <title>

### ISSUE THREAD

**Issue by zbleness** [volvo-cars/terraform-vault-backend#8]
## What is the underlying problem you would like to see solved?

Improve compatibility with terraform

## Describe the ideal solution

TVB support the HTTP `LOCK` and `UNLOCK` methods directly:
https://developer.hashicorp.com/terraform/language/settings/backends/http#lock_method

## Describe a sufficient solution

This is just a developer experience improvement.

**Comment by aekblad** [comment]
Is this a good idea? At the very least, it doesn't seem like it's worth the effort to implement WebDAV on top of FastAPI:

> Any resource that supports the LOCK method MUST, at minimum, support the XML request and response formats defined herein.

Source: [WebDAV spec, LOCK verb](https://www.rfc-editor.org/rfc/rfc4918#section-9.10)

**Comment by zbleness** [comment]
> Is this a good idea? At the very least, it doesn't seem like it's worth the effort to implement WebDAV on top of FastAPI:
> 
> > Any resource that supports the LOCK method MUST, at minimum, support the XML request and response formats defined herein.
> 
> Source: [WebDAV spec, LOCK verb](https://www.rfc-editor.org/rfc/rfc4918#section-9.10)

The goal is to play well with most recent terraform releases and maybe opentao if there is community interest. Full, RFC compliance is in scope here.

**Comment by aekblad** [comment]
> > Is this a good idea? At the very least, it doesn't seem like it's worth the effort to implement WebDAV on top of FastAPI:
> > > Any resource that supports the LOCK method MUST, at minimum, support the XML request and response formats defined herein.
> > 
> > 
> > Source: [WebDAV spec, LOCK verb](https://www.rfc-editor.org/rfc/rfc4918#section-9.10)
> 
> The goal is to play well with most recent terraform releases and maybe opentao if there is community interest. Full, RFC compliance is in scope here.

Right, I was worried about what's required for interoperability changing from under us without notice. After all, considering the specs involved, it wouldn't be a breaking change for clients to suddenly start using an XML request body to pass parameters.

However, I realize now that this _only_ affects parameters. If we just make sure to keep our endpoints free of query parameters we won't have this problem.
